src/util/amount.rs

```diff
@@ -67,16 +67,47 @@ impl FromStr for Denomination {
     type Err = ParseAmountError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "BTC" => Ok(Denomination::Bitcoin),
-            "mBTC" => Ok(Denomination::MilliBitcoin),
-            "uBTC" => Ok(Denomination::MicroBitcoin),
-            "bits" => Ok(Denomination::Bit),
-            "satoshi" => Ok(Denomination::Satoshi),
-            "sat" => Ok(Denomination::Satoshi),
-            "msat" => Ok(Denomination::MilliSatoshi),
-            d => Err(ParseAmountError::UnknownDenomination(d.to_owned())),
+        if s == "Msat" {        // So as not to confuse with mega sats.
+            return Err(ParseAmountError::UnknownDenomination(s.to_owned()));
```

Comment:
Maybe we could have `PossiblyConfusingDenomination` variant or something similar.

I'm still unsure if we should allow `MSAT`, `MSat` and similar.

Reply:
Can do, will re-spin later today.